ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,869 @@
+# Series, Series, scalar
+@execute_node.register(ops.Where, dd.Series, dd.Series, dd.Series)
+@execute_node.register(ops.Where, dd.Series, dd.Series, scalar_types)
+def execute_node_where_series_series_series(op, cond, true, false, **kwargs):
+    # No need to turn false into a series, dask will broadcast it
+    return true.where(cond, other=false)
+
+
+# Series, scalar, Series
+def execute_node_where_series_scalar_scalar(op, cond, true, false, **kwargs):
+    return dd.from_array(np.repeat(true, len(cond))).where(cond, other=false)
+
+
+# Series, scalar, scalar`
+for scalar_type in scalar_types:
+    execute_node_where_series_scalar_scalar = execute_node.register(
+        ops.Where, dd.Series, scalar_type, scalar_type
+    )(execute_node_where_series_scalar_scalar)
+
+
+# scalar, Series, Series
+@execute_node.register(ops.Where, boolean_types, dd.Series, dd.Series)
+def execute_node_where_scalar_scalar_scalar(op, cond, true, false, **kwargs):
+    # Note that it is not necessary to check that true and false are also
+    # scalars. This allows users to do things like:
+    # ibis.where(even_or_odd_bool, [2, 4, 6], [1, 3, 5])
+    return true if cond else false
+
+
+# scalar, Series, scalar
+@execute_node.register(ops.Where, boolean_types, dd.Series, scalar_types)
+def execute_node_where_scalar_series_scalar(op, cond, true, false, **kwargs):
+    if cond:
+        return true
+    else:
+        # TODO double check this is the right way to do this
+        out = dd.from_array(np.repeat(false, len(true)))
+        out.index = true.index
+        return out
+
+
+# scalar, scalar, Series
+@execute_node.register(ops.Where, boolean_types, scalar_types, dd.Series)
+def execute_node_where_scalar_scalar_series(op, cond, true, false, **kwargs):
+    return dd.from_array(np.repeat(true, len(false))) if cond else false
+
+
+@execute_node.register(DaskTable, DaskClient)
+def execute_database_table_client(
+    op, client, timecontext: Optional[TimeContext], **kwargs
+):
+    df = client.dictionary[op.name]
+    if timecontext:
+        begin, end = timecontext
+        if TIME_COL not in df:
+            raise com.IbisError(
+                f'Table {op.name} must have a time column named {TIME_COL}'
+                ' to execute with time context.'
+            )
+        # filter with time context
+        mask = df[TIME_COL].between(begin, end)
+        return df.loc[mask].reset_index(drop=True)
+    return df
+
+
+@execute_node.register(ops.IfNull, dd.Series, simple_types)
+@execute_node.register(ops.IfNull, dd.Series, dd.Series)
+def execute_node_ifnull_series(op, value, replacement, **kwargs):
+    return value.fillna(replacement)
+
+
+@execute_node.register(ops.IfNull, simple_types, dd.Series)
+def execute_node_ifnull_scalar_series(op, value, replacement, **kwargs):
+    return (
+        replacement
+        if isnull(value)
+        else dd.from_pandas(
+            pd.Series(value, index=replacement.index), npartitions=1
```

Comment:
How does this work? Isn't replace.index a distributed index? Will this trigger any data materialization?

Reply:
This is a good question - I _think_ this is done lazily

```python
In [34]: dask_base
Out[34]:
Dask Series Structure:
npartitions=1
0    datetime64[ns]
2               ...
dtype: datetime64[ns]
Dask Name: from_pandas, 1 tas
In [32]: dd.from_pandas(pd.Series(5, index=dask_base.index), npartitions=1)
Out[32]:
Dask Series Structure:
npartitions=1
0    int64
2      ...
dtype: int64
Dask Name: from_pandas, 1 tasks

In [33]: dd.from_pandas(pd.Series(5, index=dask_base.index), npartitions=1).compute()
Out[33]:
0    5
1    5
2    5
dtype: int64

```

Reply:
a lot of operations like this will trigger materialization at times. this is ok. its on the worker in a partition. its usually much easier to just do it this way.

Reply:
that is the type of the `value` and `replacement` here? Also why do we return a series of a single partition? Should this method return the same partitions as the input dd.Series?

Reply:
I should express the partitions in chunks rather than having a static number.